### PR TITLE
Do panic in integration tests if connection can't be established

### PR DESCRIPTION
**Pull request by rumyantseva**
# Description

In `integration/setup/listener.go` in the `tb.Cleanup` call we wait for a message from `done` channel. If the connection with the client wasn't established, this message will be never received, as listener is still runing. As a result, the tests are hanging until timeout. This PR fixes this problem by panicking if the connection can't be established.

For example, this problem occurred in this https://github.com/FerretDB/FerretDB/pull/2540 PR, and it was hard to catch the problem with failing integration tests.

## Readiness checklist

* [ ] I added/updated unit tests.
* [ ] I added/updated integration/compatibility tests.
* [ ] I added/updated comments and checked rendering.
* [x] I made spot refactorings.
* [ ] I updated user documentation.
* [x] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: integration/setup/client.go [hunk 1]
```diff
@@ -106,12 +106,14 @@ func setupClient(tb testing.TB, ctx context.Context, uri string) *mongo.Client {
 	defer trace.StartRegion(ctx, "setupClient").End()
 
 	client, err := makeClient(ctx, uri)
-	require.NoError(tb, err, "URI: %s", uri)
+	if err != nil {
+		return nil, err
+	}
```
Comment (AlekSi) [inline integration/setup/client.go:111]: I don't think we should assert on some errors and return others. The simplest solution might be panic

Comment (rumyantseva) [reply integration/setup/client.go:111]: Ok, if panic is enough, let's panic.

Comment (AlekSi) [reply integration/setup/client.go:111]: Actually, there is a simpler fix

**Review by w84thesun — APPROVED** [review]
Nice catch!

**Review by AlekSi — APPROVED** [review]

**Review by noisersup — APPROVED** [review]
Nice one!

**Pull request merged**